### PR TITLE
Simple Captcha faucet

### PR DESCRIPTION
**Pull request by wsdt**
:clipboard: Related to #996 

## Overview

Chain agnostic faucet.

Based on previous discussions and the now paid Twitter API, I created a super simple lambda faucet that uses HCaptcha and has a 24h limit on it. 

Lambda & API Gateway already running on AWS. 

Before we deploy this, we should integrate & test it on the running frontend. 

The lambda expects the following request: 

```json
{
 "hcaptcha_resp": "{CAPTCHA_RESPONSE_TO_VALIDATE}",
 "network": "{BOBA_BNB_TESTNET | BOBA_GOERLI_TESTNET | BOBA_AVAX_TESTNET}", 
 "wallet_addr": "{WALLET_ADDRESS_TO_SEND_FUNDS_TO}"
}
```
Private key of faucet (as always) should only contain testnet funds. 

Current issuance: 
* Alt L2s: Native amount (BOBA) = 1 BOBA, token amount (e.g. tBNB) = 0.1 
* Goerli: Eth = 0.1, Boba token = 1

## Testing
Ran test events on AWS. 

Exemplary tx on boba bnb testnet (currently two tx, one for native, one for token): 
https://blockexplorer.testnet.bnb.boba.network/tx/0xecd44711333f8fa49ddd293084bc4487e97b04e9dd7ffd1fa3f3259d79479dc2

### CODE REVIEW

File: boba_community/hc-captcha-faucet-2023/captcha.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+# or...
+# authorized_contract = '0xOF_YOUR_HELPER_CONTRACT' # to restrict access to only your smart contract
+
+# NOTE: When taking the data payload from the original event for debugging then remove the first 64 bits!
+def lambda_handler(input, context):
+  print("DEBUG: ", input)
+
+  print("DEBUG: from Geth:", input)
```
Comment (InoMurko) [inline boba_community/hc-captcha-faucet-2023/captcha.py]:
```suggestion
  print("DEBUG: from L2:", input)
```

File: boba_community/hc-captcha-faucet-2023/captcha.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        "txId": txId
+      })
+    }
+  else:
+    returnPayload = {
+      'statusCode': 400,
+      'body': json.dumps({
+        "error": error_msg
+      })
+    }
```
Comment (InoMurko) [inline boba_community/hc-captcha-faucet-2023/captcha.py]: I would refactor this - extract the validation part that either makes or breaks claiming funds once every 24 hours, and then the separate part is response.
- captcha solving
- validation
- response

File: boba_community/hc-captcha-faucet-2023/captcha.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  allowed_to_claim = time_limit_ok and check_hcaptcha(hcaptcha_client_response)
+  if not allowed_to_claim:
+    if not time_limit_ok:
+      error_msg = "Wait 24h"
+    else:
+      error_msg = "Captcha failed"
+
+  if allowed_to_claim:
+    # TBD: potentially add rpc url whitelist
```
Comment (InoMurko) [inline boba_community/hc-captcha-faucet-2023/captcha.py]: wdym?

**Review by InoMurko — APPROVED** [review]
left a few nitpicks re style, but good stuff

File: boba_community/hc-captcha-faucet-2023/captcha.py (hunk 3 above)
Comment (wsdt) [reply boba_community/hc-captcha-faucet-2023/captcha.py]: this faucet is chain agnostic, basically you can send any provider url to the faucet and issue ETH/BOBA there. 
From a security standpoint I just noted that we might want to consider a provider whitelist to ensure that only pre-defined testnets work (and under no way mainnet funds, although why should we send real funds to this PK).

Comment (wsdt) [reply boba_community/hc-captcha-faucet-2023/captcha.py]: added simple whitelist

**Review by souradeep-das — APPROVED** [review]
looks great!

**Pull request merged**